scripts/bert/finetune_classifier.py

```diff
@@ -78,26 +80,59 @@
 do_lower_case = 'uncased' in dataset
 tokenizer = FullTokenizer(vocabulary, do_lower_case=do_lower_case)
 
-model = BERTClassifier(bert, dropout=0.1)
-model.classifier.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+if args.task_name in ['STS-B']:
+    model = BERTRegression(bert, dropout=0.1)
+    model.regression.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+    loss_function = gluon.loss.L2Loss()
+else:
+    model = BERTClassifier(bert, dropout=0.1)
+    model.classifier.initialize(init=mx.init.Normal(0.02), ctx=ctx)
+    loss_function = gluon.loss.SoftmaxCELoss()
+
+
 logging.info(model)
 model.hybridize(static_alloc=True)
 
-loss_function = gluon.loss.SoftmaxCELoss()
+
 loss_function.hybridize(static_alloc=True)
 
-metric = mx.metric.Accuracy()
 
-trans = ClassificationTransform(tokenizer, MRPCDataset.get_labels(), args.max_len)
-data_train = MRPCDataset('train').transform(trans)
-data_dev = MRPCDataset('dev').transform(trans)
+tasks = {
+    'MRPC':MRPCDataset,
+    'QQP':QQPDataset,
+    'QNLI':QNLIDataset,
+    'RTE':RTEDataset,
+    'STS-B':STSBDataset,
+    'CoLA':COLADataset,
+    'MNLI':MNLIDataset,
+    'WNLI':WNLIDataset
+    }
+
+
+task = tasks[args.task_name]
+
+if args.task_name in ['STS-B',]:
+    trans = RegressionTransform(tokenizer, args.max_len)
+elif args.task_name in ['CoLA']:
+    trans = ClassificationTransform(tokenizer, task.get_labels(), args.max_len, pair=False)
+else:
+    trans = ClassificationTransform(tokenizer, task.get_labels(), args.max_len)
+
+
+if args.task_name == 'MNLI':
+    data_train = task('dev_matched').transform(trans)
+    data_dev = task('dev_mismatched').transform(trans)
+else:
+    data_train = task('train').transform(trans)
+    data_dev = task('dev').transform(trans)
+
 
 bert_dataloader = mx.gluon.data.DataLoader(data_train, batch_size=batch_size,
                                            shuffle=True, last_batch='rollover')
 bert_dataloader_dev = mx.gluon.data.DataLoader(data_dev, batch_size=test_batch_size,
                                                shuffle=False)
```

Comment:
Perhaps it would be more elegant and efficient to consider using [gluonnlp.data.batchify](http://gluon-nlp.mxnet.io/api/modules/data.batchify.html) and [gluonnlp.data.FixedBucketSampler](http://gluon-nlp.mxnet.io/api/modules/data.html#gluonnlp.data.FixedBucketSampler).

Reply:
I tested all GLUE tasks with  @eric-haibin-lin's PR and feel really fast, it needs to be merge with.